### ISSUE TITLE
`PrivateInFinalClassRule` fails with traits containing an `abstract` `protected` method implemented in a `final` class

### ISSUE THREAD

**Issue by kayw-geek** [ergebnis/phpstan-rules#838]
### Description
The `PrivateInFinalClassRule` in PHPStan is currently failing under the following scenario:

When a trait declares an abstract protected method and a final class uses this trait and implements the abstract method, PHPStan issues an error due to the implemented method being `protected` instead of `private`. However, I believe that the implemented abstract method should retain the same access modifier, which is `protected`.

### Steps to Reproduce

1. **Create a trait with an abstract protected method:**

   ```php
   trait ExampleTrait {
       abstract protected function exampleMethod();
   }
   ```

2. **Create a final class that uses the trait and implements the abstract method with a protected access modifier:**

   ```php
   final class ExampleClass {
       use ExampleTrait;

       protected function exampleMethod() {
           // implementation
       }
   }
   ```

3. **Run PHPStan analysis on the above code**

### Expected Behavior
PHPStan should not issue an error when a final class implements an abstract protected method from a trait.
